packages/jest-config/src/normalize.ts

```diff
@@ -985,10 +986,12 @@ export default function normalize(
 
   if (argv.all) {
     newOptions.onlyChanged = false;
+    newOptions.onlyFailures = false;
   } else if (newOptions.testPathPattern) {
     // When passing a test path pattern we don't want to only monitor changed
-    // files unless `--watch` is also passed.
+    // or failed files unless `--watch` is also passed.
     newOptions.onlyChanged = newOptions.watch;
+    newOptions.onlyFailures = newOptions.watch;
```

Comment:
this is a change in behavior for the `watch` flag, but I like it

Reply:
hmm, coming back to this, I don't 😅 If I run in watch mode I want to run all tests I specify, not just the ones that failed last time (unless I opt in to this behavior)

Reply:
When I said I liked it, what I thought at the time this did was, that in addition to running the specified files, we'll run any failing tests. But that's not what this does

Reply:
There is no way to opt out of this behaviour without using the command line `Press f` as far as I can tell. It's broken running jest in watch mode in webstorm as you cannot interact with the CLI. This behaviour is a pain.

Reply:
@domarmstrong reverted in #10692, will make a new release once #10690 is solved

Reply:
Brilliant thanks for the info 👍